src/Launcher.vala

```diff
@@ -28,6 +31,56 @@ public class Dock.Launcher : Gtk.Button {
         windows = new GLib.List<AppWindow> ();
         get_style_context ().add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
 
+        var pinned_menu_item = new MenuItem (null, null);
+        pinned_menu_item.set_attribute_value ("custom", "pinned-item");
+
+        var close_menu_item = new MenuItem (null, null);
+        close_menu_item.set_attribute_value ("custom", "close-item");
```

Comment:
As part of the user survey results, it doesn't seem like very many people use this feature so we intended to drop it